lib/elixir/src/elixir_quote.erl

```diff
@@ -418,7 +430,8 @@ keystore(_Key, Meta, nil) ->
 keystore(Key, Meta, Value) ->
   lists:keystore(Key, 1, Meta, {Key, Value}).
 keynew(Key, Meta, Value) ->
-  case keyfind(Key, Meta) of
-    {Key, _} -> Meta;
-    _ -> keystore(Key, Meta, Value)
+  case lists:keymember(Key, 1, Meta) of
+    true -> Meta;
+    false when Value =:= nil -> Meta;
```

Comment:
It seems we always return `Meta` when the `Value` is `nil`. When is the value `nil` in the first place?

Reply:
I can't identify any case where the value would be `nil`.

Reply:
I just inlined the code from keystore that explicitly checks for the value being `nil`. Maybe it has some other uses there, but here it is useless?

Reply:
@michalmuskala so maybe we should move the `nil` check up, like we do for `keystore`. Or remove it altogether. Your call.

Reply:
Yeah, it seems `keynew` is only ever called with either a `line` or `counter` and a value of `nil` makes no sense in there. I'll remove the check.